[dockerng] WIP No more runtime args passed to docker.start()

ticosax opened the pull request:
fixes #27661 
I open this pull request for discussion.
Some tests for `modules.dockerng` are missing.

/cc @terminalmage @alf

ticosax committed:
docker.version_info is now provided.

terminalmage commented on salt/modules/dockerng.py:
```diff
@@ -4168,174 +4304,13 @@ def signal_(name, signal):
 @_refresh_mine_cache
 @_ensure_exists
-def start(name, validate_ip_addrs=True, **kwargs):
+def start(name, **kwargs):
```
We can drop the ``**kwargs`` here, since we are no longer doing anything with them.

cachedout commented:
@ticosax This has merge conflicts. Could you please rebase it?

ticosax committed:
Do not pass any argument to docker.start

docker.create is responsible to manage all of them, since docker API 1.16

ticosax committed:
prevent potential error while reporting mismatch versions to user.

ticosax commented:
Thanks @cachedout 
@terminalmage I addressed your comment.
 When I wrote a new test I realized it is already covered. Since every extra_argument that moves from `start()` to `create()` is passed in host_config dictionary. Which support has been added by #27451.

So, if tests are passing I believe it is good to go.

terminalmage commented:
Great, I'll review it today.

ticosax commented:
a fix for the test is coming

ticosax committed:
fixup! Do not pass any argument to docker.start

ticosax commented:
@terminalmage I recommend to review a first pass with https://github.com/saltstack/salt/pull/27676/files?w=1
As big part of changes are just re-indentation.

terminalmage commented:
This looks great and the one failing test on the Cent7 side is unrelated.

ticosax commented:
@terminalmage thank you.
